### DISCUSSION TITLE
Auto-Implement Stubs for Interface Implementations

### DISCUSSION THREAD

**Discussion by yribeiro** [microsoft/pylance-release#2724] [Enhancement]
At the moment, I do not think there is an auto-complete feature for python classes that implement an abstract interface. It would be super useful to auto generate the function stubs (along with any property decorators) that are defined in the `abc.ABC` interface class.

**Comment by erictraut** [comment]
Pylance's completion provider will offer to fill in a stub implementation for a method defined in a base class if you start to type the name of that method.

For example:
```python
class Base(ABC):
    @abstractmethod
    def test(self, a: int, b: str) -> int:
        ...

class Child(Base):
    def te # Hit tab here and you get...
```

```python
class Child(Base):
    def test(self, a: int, b: str) -> int:
        return super().test(a, b)
```

**Comment by yribeiro** [comment]
That's brill @erictraut. Thanks very much. Is there no option to fill in all the stubs? PyCharm has a feature that allows you to right click the child class, and fill in the parent methods as stubs.

**Comment by judej** [comment]
Thanks @yribeiro, moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by ccrvlh** [comment]
Would be great to have this also implemented as a "light bulb", very useful for interfaces with several methods. If I'm not mistaken Typescript has something similar, or maybe C#, not sure.

**Comment by E5presso** [comment]
Is there any updates or plans? I think this feature (implementing all stubs from abc) is might useful!

**Reply by erictraut** [reply]
If you're interested in this feature, please "upvote" it by giving it a thumbs up. The pylance team uses such feedback to prioritize features. So far, only one upvote has been received for this enhancement request, so it's much lower priority than others.

**Reply by dsinghvi** [reply]
@erictraut looks like the upvotes have picked up. How many votes do you need to push up the priority?

**Reply by rchiodo** [reply]
Priority is based on a lot of stuff. Up votes is just one of the things we use to determine when to implement things. Relative priority for this discussion item is 18th in the list I believe.

**Comment by judej** [comment]
This has been added to our backlog. Use #5362 to track the progress

**Reply by yribeiro** [reply]
Amazing! Thank you.
